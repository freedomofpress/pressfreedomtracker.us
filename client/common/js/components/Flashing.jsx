import React from 'react'
import PropTypes from 'prop-types'

import './Flashing.sass'

export default function Flashing({ flashing = true, children }) {
	return <div className={flashing ? 'flashing' : 'flashing flashing--off'}>{children}</div>
}

Flashing.propTypes = {
	flashing: PropTypes.bool,
	children: PropTypes.node.isRequired,
}
