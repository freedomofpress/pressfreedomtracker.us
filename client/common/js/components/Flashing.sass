@keyframes flashing
	from
		opacity: 0.25
	to
		opacity: 0.75

.flashing
	animation: flashing 1s infinite ease-in-out alternate-reverse

.flashing--off
	animation: none
