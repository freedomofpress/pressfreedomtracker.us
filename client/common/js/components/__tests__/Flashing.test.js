/* eslint-disable no-undef,react/jsx-filename-extension */
import React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow'
import Flashing from '../Flashing'

test('renders Flashing true', () => {
	const renderer = new ShallowRenderer()
	expect(renderer.render(
		<Flashing flashing>
			test
		</Flashing>,
	)).toMatchSnapshot()
})

test('renders Flashing false', () => {
	const renderer = new ShallowRenderer()
	expect(renderer.render(
		<Flashing flashing={false}>
			test
		</Flashing>,
	)).toMatchSnapshot()
})
