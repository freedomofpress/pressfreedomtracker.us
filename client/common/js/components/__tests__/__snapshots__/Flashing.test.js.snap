// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Flashing false 1`] = `
<div
  className="flashing flashing--off"
>
  test
</div>
`;

exports[`renders Flashing true 1`] = `
<div
  className="flashing"
>
  test
</div>
`;
